app/download.go

```diff
@@ -34,10 +36,20 @@ func (a *App) DownloadFromURL(downloadURL string) ([]byte, error) {
 	client := a.HTTPService().MakeClient(true)
 	client.Timeout = HTTP_REQUEST_TIMEOUT
 
-	resp, err := client.Get(downloadURL)
+	var resp *http.Response
+	err = utils.ProgressiveRetry(func() error {
+		resp, err = client.Get(downloadURL)
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch from %s", downloadURL)
+		}
+
+		return nil
+	})
+
```

Comment:
If we are adding a gap after the `err != nil` check, then let's remove the gap on this line. Error checks usually go together with their calling functions.